shared/src/commonMain/kotlin/org/hyperskill/app/profile/presentation/ProfileReducer.kt

```diff
@@ -289,7 +301,16 @@ class ProfileReducer : StateReducer<State, Message, Action> {
         message: Message.BadgeClicked
     ): ReducerResult =
         if (state is State.Content) {
+            val clickedBadge = state.badgesState.badges.firstOrNull { it.kind == message.badgeKind }
+            val showAction = Action.ViewAction.ShowBadgeDetailsModal(
+                if (clickedBadge == null) {
+                    Action.ViewAction.BadgeDetails.Kind(message.badgeKind)
+                } else {
+                    Action.ViewAction.BadgeDetails.Badge(clickedBadge)
+                }
+            )
```

Comment:
1. How is it even possible that a user clicks on a badge that is not in the state?
2. Why do we need `ProfileFeature.Action.ViewAction.BadgeDetails`?

Maybe I'm missing smth, but this is what I'm expecting:
```kotlin
val clickedBadge = state.badgesState.badges.firstOrNull { it.kind == message.badgeKind }
val showAction = if (clickedBadge != null) {
    Action.ViewAction.ShowBadgeModal(clickedBadge)
} else {
    null
}

state to setOfNotNull(
    showAction,
    Action.LogAnalyticEvent(
        ProfileClickedBadgeCardHyperskillAnalyticsEvent(
            badgeKind = message.badgeKind,
            isLocked = message.badgeKind !in state.badgesState.badges.map { it.kind }
        )
    )
)
```

Reply:
@ivan-magda 
> How is it even possible that a user clicks on a badge that is not in the state?

In some cases, backend doesn't return locked badges because of the [bug](https://vyahhi.myjetbrains.com/youtrack/issue/ALT-8526/Badges-in-a-tooltip-are-not-visible-in-some-cases).
In these cases, locked badges are added to the unlocked in the ViewState mapper.
So user can click on the locked badge which is not a part of the backend response and it is not in the state.

Reply:
@ivan-magda 
> Why do we need ProfileFeature.Action.ViewAction.BadgeDetails?

It is used to handle clicking on the usual badge (returned by the backend).
`ProfileFeature.Action.ViewAction.Kind` is used to handle clicking on the badge which is not a part of the state (see previous comment).

Reply:
@XanderZhu OK, without knowing implementation details it's super unclear.

So I propose you to:
1. Choose a better naming for the `ProfileFeature.Action.ViewAction.BadgeDetails` cases
2. Add helper comments to `BadgesViewStateMapper` according to this https://github.com/hyperskill/mobile-app/pull/607#discussion_r1286858364
3. Add helper comments to badge click handling according to this https://github.com/hyperskill/mobile-app/pull/607#discussion_r1286861378